spring-cloud-starter-stream-processor-httpclient/src/main/java/org/springframework/cloud/stream/app/httpclient/processor/HttpclientProcessorFunctionConfiguration.java

```diff
@@ -0,0 +1,107 @@
+import java.util.function.Function;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.messaging.Message;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * A {@link Function} that makes requests to an HTTP resource and emits the
+ * response body as a message payload.
+ *
+ * @author Waldemar Hummer
+ * @author Mark Fisher
+ * @author Gary Russell
+ * @author Christian Tzolov
+ * @author David Turanski
+ **/
+@Configuration
+@EnableConfigurationProperties(HttpclientProcessorProperties.class)
+public class HttpclientProcessorFunctionConfiguration {
+
+	public static final String HTTPCLIENT_PROCESSOR_FUNCTION_NAME = "httpRequest";
+
+	private static final Log LOG = LogFactory.getLog(HttpclientProcessorConfiguration.class);
+
+	@Bean
+	public RestTemplate restTemplate() {
+		return new RestTemplate();
+	}
+
+	@Bean(name = HTTPCLIENT_PROCESSOR_FUNCTION_NAME)
+	public Function<Message<?>, Object> httpRequest(RestTemplate restTemplate,
+		HttpclientProcessorProperties properties) {
+
+		return message -> {
+			try {
+				/* construct headers */
+				HttpHeaders headers = new HttpHeaders();
+				if (properties.getHeadersExpression() != null) {
+					Map<?, ?> headersMap = properties.getHeadersExpression().getValue(message, Map.class);
+					for (Map.Entry<?, ?> header : headersMap.entrySet()) {
+						if (header.getKey() != null && header.getValue() != null) {
+							headers.add(header.getKey().toString(),
+								header.getValue().toString());
+						}
+					}
+				}
+
+				Class<?> responseType = properties.getExpectedResponseType();
+				HttpMethod method = null;
+				if (properties.getHttpMethodExpression() != null) {
+					method = properties.getHttpMethodExpression().getValue(message, HttpMethod.class);
+				}
+				else {
+					method = properties.getHttpMethod();
+				}
+				String url = properties.getUrlExpression().getValue(message, String.class);
+				Object body = null;
+				if (properties.getBody() != null) {
+					body = properties.getBody();
+				}
+				else if (properties.getBodyExpression() != null) {
+					body = properties.getBodyExpression().getValue(message);
+				}
+				else {
+					body = message.getPayload();
+				}
+				URI uri = new URI(url);
+				RequestEntity<?> request = new RequestEntity<>(body, headers, method, uri);
+				ResponseEntity<?> response = restTemplate.exchange(request, responseType);
+				return properties.getReplyExpression().getValue(response);
+			}
+			catch (Exception e) {
+				LOG.warn("Error in HTTP request", e);
+				return null;
```

Comment:
Why `return null`? Why just not re-throw?
I know this is an existing code, but still...